front/src/reducers/main/index.ts

```diff
@@ -0,0 +1,88 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Notification } from 'types';
+
+export interface MainState {
+  loading: number;
+  notifications: Array<Notification>;
+  lastInterfaceVersion: string;
+}
+
+export const mainInitialState: MainState = {
+  // Number of running task
+  loading: 0,
+  // errors
+  notifications: [],
+  lastInterfaceVersion: '',
+};
+
+const mainSlice = createSlice({
+  name: 'main',
+  initialState: mainInitialState,
+  reducers: {
+    setLoading(state) {
+      state.loading += 1;
+    },
+    setSuccess(state, action: PayloadAction<Notification | undefined>) {
+      state.loading = state.loading > 0 ? state.loading - 1 : 0;
+      if (action.payload) {
+        state.notifications.push({
+          type: action.payload.type || 'success',
+          title: action.payload.title,
+          text: action.payload.text,
+          date: action.payload.date || new Date(),
+        });
+      }
+    },
```

Comment:
We should type action as { title: string, message: string } (do not allow undefined or object without title or text). On the contrary, the actionType and date are not needed, so you can just remove it from the type.
It should be the same for the others notifications.
```suggestion
    setSuccess(state, action: PayloadAction<{ title: string, message: string}>) {
      state.loading = state.loading > 0 ? state.loading - 1 : 0;
      if (action.payload) {
        state.notifications.push({
          type: 'success',
          title: action.payload.title,
          text: action.payload.text,
          date: new Date(),
        });
      }
    },
```

Reply:
I agree with you, but there in reducers/editors.ts file, setSuccess is called without a payload. Maybe we can pass on something like this  `{ title: 'successful loading data model, text: '' }` as parameter ?